ApiDemos/kotlin/app/src/main/java/com/example/kotlindemos/PolylineDemoActivity.kt

```diff
@@ -0,0 +1,291 @@
+    private val MAX_WIDTH_PX = 100
+    private val MAX_HUE_DEGREES = 360
+    private val MAX_ALPHA = 255
+    private val PATTERN_DASH_LENGTH_PX = 50
+    private val PATTERN_GAP_LENGTH_PX = 20
+
+    // City locations for mutable polyline.
+    private val adelaideLatLng = LatLng(-34.92873, 138.59995)
+    private val darwinLatLng = LatLng(-12.4258647, 130.7932231)
+    private val melbourneLatLng = LatLng(-37.81319, 144.96298)
+    private val perthLatLng = LatLng(-31.95285, 115.85734)
+
+    // Airport locations for geodesic polyline.
+    private val aklLatLng = LatLng(-37.006254, 174.783018)
+    private val jfkLatLng = LatLng(40.641051, -73.777485)
+    private val laxLatLng = LatLng(33.936524, -118.377686)
+    private val lhrLatLng = LatLng(51.471547, -0.460052)
+
+    private val dot = Dot()
+    private val dash = Dash(PATTERN_DASH_LENGTH_PX.toFloat())
+    private val gap = Gap(PATTERN_GAP_LENGTH_PX.toFloat())
+    private val patternDotted = Arrays.asList(dot, gap)
+    private val patternDashed = Arrays.asList(dash, gap)
+    private val patternMixed = Arrays.asList(dot, gap, dot, dash, gap)
+
+    private lateinit var mutablePolyline: Polyline
+    private lateinit var hueBar: SeekBar
+    private lateinit var alphaBar: SeekBar
+    private lateinit var widthBar: SeekBar
+    private lateinit var startCapSpinner: Spinner
+    private lateinit var endCapSpinner: Spinner
+    private lateinit var jointTypeSpinner: Spinner
+    private lateinit var patternSpinner: Spinner
+    private lateinit var clickabilityCheckbox: CheckBox
+
+    // These are the options for polyline caps, joints and patterns. We use their
+    // string resource IDs as identifiers.
+    private val capTypeNameResourceIds = intArrayOf(R.string.cap_butt, // Default
+            R.string.cap_round, R.string.cap_square, R.string.cap_image)
+
+    private val jointTypeNameResourceIds = intArrayOf(R.string.joint_type_default, // Default
+            R.string.joint_type_bevel, R.string.joint_type_round)
+
+    private val patternTypeNameResourceIds = intArrayOf(R.string.pattern_solid, // Default
+            R.string.pattern_dashed, R.string.pattern_dotted, R.string.pattern_mixed)
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.polyline_demo)
+
+        hueBar = findViewById<SeekBar>(R.id.hueSeekBar).apply {
+            max = MAX_HUE_DEGREES
+            progress = 0
+        }
+
+        alphaBar = findViewById<SeekBar>(R.id.alphaSeekBar).apply {
+            max = MAX_ALPHA
+            progress = MAX_ALPHA
+        }
+
+        widthBar = findViewById<SeekBar>(R.id.widthSeekBar).apply {
+            max = MAX_WIDTH_PX
+            progress = MAX_WIDTH_PX / 2
+        }
+
+        startCapSpinner = findViewById<Spinner>(R.id.startCapSpinner).apply {
+            adapter = ArrayAdapter(
+                    this@PolylineDemoActivity, android.R.layout.simple_spinner_item,
```

Comment:
I don't think you need a new line after ArrayAdapter(

Reply:
Done!